supernode/daemon/mgr/task/manager_util_test.go

```diff
@@ -65,19 +62,22 @@ func (s *TaskUtilTestSuite) TearDownSuite(c *check.C) {
 	s.contentLengthStub.Reset()
 }
 
-func (s *TaskUtilTestSuite) TestAddOrUpdateTask(c *check.C) {
+func (s *TaskUtilTestSuite) TestEqualsTask(c *check.C) {
```

Comment:
I am curious that why we change this test? I think you can add `TestEqualsTask` and keep the previous  `TestAddOrUpdateTask`

Reply:
In fact, I have been having a doubt for that when should we add unit tests for functions before until I saw [this article](https://juejin.im/entry/5bc00fb0f265da0af33498c1).

According to this article,I think `AddOrUpdateTask` is a function which is complicates and with many external dependencies. So we should not test this function directly but divide it into two parts: 
1. Handle complex logical parts 
2. Process dependent parts
And then write the unit testing for part one.